src/BenchmarkDotNet/Analysers/Conclusion.cs

```diff
@@ -11,36 +11,45 @@ public sealed class Conclusion : IEquatable<Conclusion>
         public string AnalyserId { get; }
 
         public ConclusionKind Kind { get; }
+        
+        public bool Mergeable { get; }
 
         [NotNull]
         public string Message { get; }
 
         [CanBeNull]
         public BenchmarkReport Report { get; }
 
-        private Conclusion([NotNull] string analyserId, ConclusionKind kind, [NotNull] string message, [CanBeNull] BenchmarkReport report)
+        private Conclusion([NotNull] string analyserId, 
+                           ConclusionKind kind, 
+                           [NotNull] string message, 
+                           [CanBeNull] BenchmarkReport report, 
+                           bool mergeable)
         {
             AnalyserId = analyserId;
             Kind = kind;
             Message = message;
             Report = report;
+            Mergeable = mergeable;
         }
 
-        public static Conclusion CreateHint(string analyserId, string message, [CanBeNull] BenchmarkReport report = null) 
-            => new Conclusion(analyserId, ConclusionKind.Hint, message, report);
+        public static Conclusion CreateHint(string analyserId, string message, [CanBeNull] BenchmarkReport report = null, bool mergeable = true) 
+            => new Conclusion(analyserId, ConclusionKind.Hint, message, report, mergeable);
 
-        public static Conclusion CreateWarning(string analyserId, string message, [CanBeNull] BenchmarkReport report = null) 
-            => new Conclusion(analyserId, ConclusionKind.Warning, message, report);
+        public static Conclusion CreateWarning(string analyserId, string message, [CanBeNull] BenchmarkReport report = null, bool mergeable = true) 
+            => new Conclusion(analyserId, ConclusionKind.Warning, message, report, mergeable);
 
-        public static Conclusion CreateError(string analyserId, string message, [CanBeNull] BenchmarkReport report = null) 
-            => new Conclusion(analyserId, ConclusionKind.Error, message, report);
+        public static Conclusion CreateError(string analyserId, string message, [CanBeNull] BenchmarkReport report = null, bool mergeable = true) 
+            => new Conclusion(analyserId, ConclusionKind.Error, message, report, mergeable);
 
         public bool Equals(Conclusion other)
```

Comment:
`GetHashCode` should be also updated because the `Distinct()` call use a hash set internally. In the current implementation, BenchmarkDotNet still (randomly) merges some ZeroMeasurement-conclusions.

Reply:
Thank you! Totally forgot about hashcode.